src/webgl/shaders/lighting.glsl

```diff
@@ -119,8 +119,8 @@ vec3 calculateImageDiffuse( vec3 vNormal, vec3 vViewPosition ){
 
 vec3 calculateImageSpecular( vec3 vNormal, vec3 vViewPosition ){
   vec3 worldCameraPosition =  vec3(0.0, 0.0, 0.0);
-  vec3 worldNormal = normalize(vNormal);
-  vec3 lightDirection = normalize( vViewPosition - worldCameraPosition );
+  vec3 worldNormal = normalize(vNormal * uCameraMatrix);
+  vec3 lightDirection = normalize( vViewPosition * uCameraMatrix - worldCameraPosition );
   vec3 R = reflect(lightDirection, worldNormal);
   vec2 newTexCoor = mapTextureToNormal( R );
```

Comment:
I noticed something kinda interesting. When you rotate around 360 degrees, you can always see the sun in the reflection, even though presumably at some point it should be behind the sphere and not in the reflection:

https://github.com/processing/p5.js/assets/5315059/aaad26a8-00a5-49d2-a854-6ddaad42e787

I wonder if this is a bug in `mapTextureToNormal` that we didn't notice before because we couldn't rotate the spheremap texture like we can now. Is that something you're interested in looking into as part of this change? The idea is to convert a normal into a spherical coordinate (an azimuthal angle and a polar angle, and a fixed radius of 1), and then map those angles to 2D texture coordinates on the input image. Maybe we're doing that slightly incorrectly?